daemon/c_vol.c

```diff
@@ -763,10 +805,63 @@ c_vol_mount_image(c_vol_t *vol, const char *root, const mount_entry_t *mntent)
 		}
 	}
 
-	dev = loopdev_create_new(&fd, img, 0, 0);
-	IF_NULL_GOTO(dev, error);
+	if (verityfs) {
+		TRACE("Creating verityfs");
+		char *label = mem_printf("%s-%s", uuid_string(container_get_uuid(vol->container)),
+					 mount_entry_get_img(mntent));
+		char *verity = verityfs_get_device_path_new(label);
+		if (file_is_blk(verity)) {
+			INFO("Using existing mapper device: %s", verity);
+		} else {
+			const char *root_hash = mount_entry_get_verity_sha256(mntent);
+			img_hash = c_vol_hash_image_path_new(vol, mntent);
+			IF_NULL_GOTO(img_hash, error);
+
+			int ret = verityfs_create_blk_dev(label, img, img_hash, root_hash);
```

Comment:
we do not need ret here, just directly use "if (vertityfs_create_blk_dev(...)) {"

Reply:
Done